src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -910,6 +961,17 @@ bool clang_cpp_convertert::get_decl_ref(
 
     break;
   }
+  case clang::Decl::Field:
```

Comment:
Is this repeated from the C code?

Reply:
> For `clang::Decl::Var` or `clang::Decl::Field`, yes. But for `clang::Decl::CXXConstructor`, no. E.g. We do some extra parsing for function parameters when parsing CXXCtor decl reference.
> 
> `clang_c_convertert::get_decl_ref` just contains two cases - it only distinguishes `enum` from `value decl`:
> 
> https://github.com/esbmc/esbmc/blob/4a5b329ac140257715d2e575733583b00b530532/src/clang-c-frontend/clang_c_convert.cpp#L2203-L2205
> 
> 
> https://github.com/esbmc/esbmc/blob/4a5b329ac140257715d2e575733583b00b530532/src/clang-c-frontend/clang_c_convert.cpp#L2216-L2218
> 
> Not sure how this partation would handle CXX decl kind. So I'm currently doing an experiment: In commit [2929e5b](https://github.com/esbmc/esbmc/commit/2929e5b7d22456d7f7100c2817481a0aead5b146), I'm defering everything else to `clang_c_convertert::get_decl_ref` in the `default` case.

Yeah, that makes sense to me, let's handle the `clang::Decl::CXXConstructor` and forward the rest to the

Reply:
> For `clang::Decl::Var` or `clang::Decl::Field`, yes. But for `clang::Decl::CXXConstructor`, no. E.g. We do some extra parsing for function parameters when parsing CXXCtor decl reference.
> 
> `clang_c_convertert::get_decl_ref` just contains two cases - it only distinguishes `enum` from `value decl`:
> 
> https://github.com/esbmc/esbmc/blob/4a5b329ac140257715d2e575733583b00b530532/src/clang-c-frontend/clang_c_convert.cpp#L2203-L2205
> 
> 
> https://github.com/esbmc/esbmc/blob/4a5b329ac140257715d2e575733583b00b530532/src/clang-c-frontend/clang_c_convert.cpp#L2216-L2218
> 
> Not sure how this partation would handle CXX decl kind. So I'm currently doing an experiment: In commit [2929e5b](https://github.com/esbmc/esbmc/commit/2929e5b7d22456d7f7100c2817481a0aead5b146), I'm defering everything else to `clang_c_convertert::get_decl_ref` in the `default` case.

Yeah, that makes sense to me! Let's handle the constructor case in the C++ converter and forward the rest to the C one.

Reply:
[CI regression results](https://github.com/esbmc/esbmc/actions/runs/2493590744) after forwarding looks ok.